Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -82,6 +82,36 @@ u64 DiscContent::GetSize() const
   return m_size;
 }
 
+bool DiscContent::Read(u64* offset, u64* length, u8** buffer, const DiscIO::Volume* volume) const
```

Comment:
There's a lot of duplicated code between the two variants of `DiscContent::Read`. Can't we have just one variant, which takes a volume pointer?

Probably also best to name the volume pointer `wrapped_volume` for clarity.